features/step_definitions/book-index-page.rb

```diff
@@ -1,40 +1,43 @@
 
-Given(/^there are (\d+) books in the database$/) do |arg1|
-    expect(books).to eq 100
-end
+Given(/^there are (\d+) books in the database$/) do |amount|
+  amount.to_i.times { create(:book) }
+end 
 
 When(/^I visit the root url$/) do
-  pending # express the regexp above with the code you wish you had
+  visit('/') 
 end
 
 Then(/^I see a list of books in the database$/) do
-  pending # express the regexp above with the code you wish you had
+  expect(page).to have_content(Book.first.title) 
 end
 
 Then(/^the books are ordered by published date$/) do
-  pending # express the regexp above with the code you wish you had
+  expect(page).to have_content(Book.first.published_date) 
 end
 
-Then(/^the list of (\d+) books are paginated in pages of (\d+) books per page$/) do |arg1, arg2|
-  pending # express the regexp above with the code you wish you had
+Then(/^the list of (\d+) books are paginated in pages of (\d+) books per page$/) do |total, per_page|
+  expect(page).to have_selector("tr", count: per_page.to_i + 1)  
```

Comment:
I often like to use `.has_css?` instead. You can get more specific since your typically give your elements class names. In your tests, it's not obvious what `tr` is but if it was `.book-row` or whatever, it would be a little easier to recognize. For instance:
`page.has_css?('.book-row', :count => 4)`